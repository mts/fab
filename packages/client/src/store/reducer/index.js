import { combineReducers } from 'redux'

export const getAppReducer = () =>
  combineReducers({
    context: (state = {}, action) => {
      switch (action.type) {
        default:
          return state
      }
    },
    api: (state = {}, action) => {
      switch (action.type) {
        default:
          return state
      }
    },
    ui: (state = {}, action) => {
      switch (action.type) {
        default:
          return state
      }
    },
  })
