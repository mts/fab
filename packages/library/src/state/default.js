import {
  windowObjectExists,
  isRunningOnLocalHostViaDomain as isRunningOnLocalHostViaDomainDefault,
  isRunningOnGitHubPages as isRunningOnGitHubPagesDefault,
} from '../environment'

export function getAppDefaultStateFromWindow() {
  return windowObjectExists ? window.appDefaultState : {}
}

export function getAppDefaultStateFromData({ isRunningOnLocalHost, isRunningOnGitHubPages, isRunningOnHeroku }) {
  return {
    context: {
      isRunningOnLocalHost: isRunningOnLocalHost || isRunningOnLocalHostViaDomainDefault,
      isRunningOnGitHubPages: isRunningOnGitHubPages || isRunningOnGitHubPagesDefault,
      isRunningOnHeroku,
    },
    api: {
      rest: {
        github: {},
      },
      graphql: {
        github: {},
      },
    },
    ui: {},
  }
}
