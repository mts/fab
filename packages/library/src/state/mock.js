export const mockAppState = {
  context: {
    isRunningOnLocalHost: true,
    isRunningOnGitHubPages: false,
    isRunningOnHeroku: false,
  },
  api: {
    rest: {
      github: {},
    },
  },
  ui: {},
}
