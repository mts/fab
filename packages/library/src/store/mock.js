import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { mockAppState } from '../state/mock'

export const getAppMockStore = customAppState => configureStore([thunk])(customAppState || mockAppState)
