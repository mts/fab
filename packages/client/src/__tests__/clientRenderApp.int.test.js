import { defaultRender } from './clientRenderApp.int.render'
import { getAppMockStore } from '../../../library/src/store/mock'
import { mockAppState } from '../../../library/src/state/mock'

jest.mock('react-dom', () => ({ hydrate: () => {} }))
jest.mock('../store/client', () => ({ clientAppStore: {} }))

const storeClientMockObject = require('../store/client')

describe('<App />', () => {
  beforeEach(() => {
    storeClientMockObject.clientAppStore = getAppMockStore(mockAppState)
  })

  afterEach(() => jest.clearAllMocks())

  describe('Snaphot', () => {
    test('must match defaultRender', () => {
      expect(global.renderToJSON(defaultRender)).toMatchSnapshot()
    })
  })
})
