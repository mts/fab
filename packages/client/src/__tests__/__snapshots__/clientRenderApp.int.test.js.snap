// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Snaphot must match defaultRender 1`] = `null`;
